salt/etcd-proxy/init.sls

```diff
@@ -1,6 +1,15 @@
 include:
   - etcd
 
+/etc/systemd/system/etcd.service.d:
+  file :
+    - directory
+
+
+/etc/systemd/system/etcd.service.d/etcd.service:
+  file.managed:
+    - source: salt://etcd-proxy/etcd.service
+
```

Comment:
(as a completely ignorable suggestion) I think you could write this in a more compact form:

```
/etc/systemd/system/etcd.service.d/etcd.service:
  file.directory:
    - name: /etc/systemd/system/etcd.service.d
  file.managed:
    - source: salt://etcd-proxy/etcd.service
```